x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -150,21 +149,29 @@ func (k msgServer) updateSupplier(
 	// still include Suppliers that no longer provide the services they removed.
 	// For the same reason, any SupplierEndpoint change should take effect at the
 	// beginning of the next session.
-	servicesActivationHeight := make(map[string]uint64)
+	ServicesActivationHeightMap := make(map[string]uint64)
 	for _, serviceConfig := range msg.Services {
-		servicesActivationHeight[serviceConfig.Service.Id] = uint64(sessionEndHeight)
+		ServicesActivationHeightMap[serviceConfig.Service.Id] = uint64(nextSessionStartHeight)
 		// If the service has already been staked for, keep its activation height.
 		for _, existingServiceConfig := range supplier.Services {
 			if existingServiceConfig.Service.Id == serviceConfig.Service.Id {
-				existingServiceActivationHeight := supplier.ServicesActivationHeight[serviceConfig.Service.Id]
-				servicesActivationHeight[serviceConfig.Service.Id] = existingServiceActivationHeight
+				existingServiceActivationHeight := supplier.ServicesActivationHeightsMap[serviceConfig.Service.Id]
+				ServicesActivationHeightMap[serviceConfig.Service.Id] = existingServiceActivationHeight
 				break
 			}
 		}
 	}
 
 	supplier.Services = msg.Services
-	supplier.ServicesActivationHeight = servicesActivationHeight
+	supplier.ServicesActivationHeightsMap = ServicesActivationHeightMap
 
 	return nil
 }
+
+// getNextSessionStartHeight returns the current height's next session start height.
+func getNextSessionStartHeight(ctx context.Context, sharedParams *sharedtypes.Params) int64 {
```

Comment:
OPTIONAL NIT: Wdyt of beginning to have functions like this be receiver functions of `sharedtypes.Params` so they're easier to reuse?

Reply:
I'd rather have it at `x/shared/session.go` and pass in the query height.

The current `getNextSessionStartHeight` uses `sdkCtx` to get the current height which makes it non-portable to non `on-chain use cases`.

`GetNextSessionStartHeight(sharedParams *sharedtypes.Params, queryHeight int64) int64`